### PR TITLE
Add database, factories, simple PySide6 interface

### PR DESCRIPTION
**Pull request by Rafalkufel**
* `sqlite` db
* `sqlalchemy` models
* `factories`
* Simple `PySide6` interface

### CODE REVIEW

File: app/config/configuration.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import os
+from functools import lru_cache
+from pathlib import Path
+
+from pydantic import BaseSettings
+
+
+class Settings(BaseSettings):
+    DB_FILE_NAME: str = "Alinka.db"
+    BASE_PATH: Path = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
Comment (w1stler) [inline app/config/configuration.py]: output is `str`, not `Path`:
```suggestion
    BASE_PATH: str = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```

File: app/db/queries.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        return [DecissionDbSchema.from_orm(decission) for decission in decissions]
+
+
+def get_decission_data_from_db(decision_id: int) -> DecissionDbSchema:
+    with db_session() as db:
+        decision = db.query(Decission).filter(Decission.id == decision_id).one()
+        return DecissionDbSchema.from_orm(decision)
+
+
+def create_decission_in_db(decision_data: dict):
```
Comment (w1stler) [inline app/db/queries.py]:
```suggestion
def create_decission_in_db(decision_data: dict) -> DecissionDbSchema:
```

File: app/db/queries.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    with db_session() as db:
+        decision = Decission(**decision_data)
+        db.add(decision)
+        db.commit()
+        return DecissionDbSchema.from_orm(decision)
+
+
+def update_decision_in_db(decision_id: int, decision_data: dict):
```
Comment (w1stler) [inline app/db/queries.py]:
```suggestion
def update_decision_in_db(decision_id: int, decision_data: dict) -> DecissionDbSchema::
```

File: app/docx/__init__.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from config import settings
+from constants import DocumentsTypes
+from db.queries import get_decission_data_from_db
+from docx.generate_document import Documents
+from docx.utils import convert_raw_documents_data
+
+
+def generate_documents(record_id: str) -> bool:
```
Comment (w1stler) [inline app/docx/__init__.py]:
```suggestion
def generate_documents(record_id: str) -> None:
```
I think it's not returning anything.

File: app/schemas.py [hunk 5]
```diff
@@ -37,29 +96,40 @@ def calculate_full_name(cls, values):
...
     @root_validator
     def calculate_student_description_genetive(cls, values):
         values["student_description_genetive"] = "ucznia" if values["student"] else "dziecka"
         return values
 
+    @root_validator
+    def calculate_date_of_birth(cls, values):
+        if values["birth_date"]:
+            return values
+        pesel = values["pesel"]
```
Comment (w1stler) [inline app/schemas.py:115]: Use `.get`, key might not be there.

File: app/schemas.py [hunk 6]
```diff
@@ -37,29 +96,40 @@ def calculate_full_name(cls, values):
...
-    student_description_genetive: str | None
+    student_description_genetive: str | None = Field(None, description="'ucznia' or 'dziecka'")
 
     @root_validator
     def calculate_student_description_genetive(cls, values):
         values["student_description_genetive"] = "ucznia" if values["student"] else "dziecka"
         return values
 
+    @root_validator
+    def calculate_date_of_birth(cls, values):
```
Comment (w1stler) [inline app/schemas.py]: Add typing

File: app/tests/conftest.py [hunk 7]
```diff
@@ -1,9 +1,38 @@
...
+    mock_db_session.start()
+
+
+def pytest_runtest_teardown(item):
+    from db.models import Base
+
+    with db_session() as db:
+        for table in reversed(Base.metadata.sorted_tables):
+            db.execute(table.delete())
+            db.commit()
```
Comment (w1stler) [inline app/tests/conftest.py]: Just asking - `db.commit()` is run multiple times here, is it intentional? I am just thinking that `db.commit()` could be done outside of the loop.

File: pyproject.toml [hunk 8]
```diff
@@ -1,19 +1,23 @@
 [tool.poetry]
 name = "alinka-pyside"
-version = "0.1.0"
+version = "0.2.0"
 description = "Aplication for Supporting Center for Children"
```
Comment (w1stler) [inline pyproject.toml]: Application

File: app/tests/factories.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                [
+                    Reason.AUTYZM,
+                    Reason.LEKKIE,
+                    Reason.NIESLYSZACE,
+                    Reason.NIEWIDZACE,
+                    Reason.RUCHOWA,
+                    Reason.SLABOSLYSZACE,
+                    Reason.SLABOWIDZACE,
+                ],
+                lenght=number_of_reasons,
```
Comment (w1stler) [inline app/tests/factories.py]: length

File: app/docx/generate_document.py [hunk 10]
```diff
@@ -1,16 +1,18 @@
...
 
-from docx.constants import DocumentsTypes
-from docx.schemas import DocumentData
+from constants import DocumentsTypes
 from jinja2 import Environment, FileSystemLoader
+from schemas import DocumentData
 
 TEMPLATES_BASE_PATH = os.path.join(os.path.dirname(__file__), "templates")
 loader = FileSystemLoader(TEMPLATES_BASE_PATH)
 enviroment = Environment(loader=loader)
```
Comment (w1stler) [inline app/docx/generate_document.py]: environment

**Review by w1stler — COMMENTED** [review]
Some minor comments, also change `decission` to `decision` in multiple places. 

Overall good job! 👍🏻

File: app/schemas.py (hunk 5 above)
Comment (Rafalkufel) [reply app/schemas.py:115]: You will always have `birth_date` here. Either `None` or with some value. The same with `pesel`.

File: app/tests/conftest.py (hunk 7 above)
Comment (Rafalkufel) [reply app/tests/conftest.py]: Good spot!

**Pull request merged**